[Lambda] Add operation to list functions, list function version and to delete functions

Nyholm opened the pull request:
Im creating a small app to delete layer functions not used.

jderusse commented on src/Service/Lambda/src/Enum/Runtime.php:
```diff
@@ -3,7 +3,7 @@
 
 /**
- * A runtime identifier. For example, `go1.x`.
+ * The runtime environment for the Lambda function.
```
why isn't it updated by the bot?

Nyholm replied:
Maybe it is. Let's see tomorrow when the bot runs next time.

Nyholm replied:
hm.. No update by the bot..

Strange.

jderusse replied:
Ok, I know why.
We don't have documentation for Shapes. We only have documentation when a Shape is referenced by an attribute.

For the Runtime Shape, we have differents documentations:
```
"Runtime": {
      "base": null,
      "refs": {
        "CompatibleRuntimes$member": null,
        "CreateFunctionRequest$Runtime": "<p>The identifier of the function's <a href=\"https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html\">runtime</a>.</p>",
        "FunctionConfiguration$Runtime": "<p>The runtime environment for the Lambda function.</p>",
        "ListLayerVersionsRequest$CompatibleRuntime": "<p>A runtime identifier. For example, <code>go1.x</code>.</p>",
        "ListLayersRequest$CompatibleRuntime": "<p>A runtime identifier. For example, <code>go1.x</code>.</p>",
        "UpdateFunctionConfigurationRequest$Runtime": "<p>The identifier of the function's <a href=\"https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html\">runtime</a>.</p>"
      }
    },
```
That's why: before your PR, the object was only used by `FunctionConfiguration`.
Now it's used by `FunctionConfiguration` and `ListLayerVersionsRequest` and the second definition is used.

Nyholm commented:
The PR is updated and ready for review. 

I generated 3 operations and added tests. I also added a changelog as I prepare a release after merging this.

jderusse commented on src/Service/Lambda/tests/Integration/LambdaClientTest.php:
```diff
@@ -38,6 +41,20 @@ public function testAddLayerVersionPermission(): void
 
+    public function testDeleteFunction(): void
+    {
+        self::markTestSkipped('The Lambda Docker image does not implement DeleteFunction.');
```
ho. Really? :(

Nyholm replied:
I got a 405
